### PR TITLE
$xavi(...) 

### PR DESCRIPTION
**Pull request by linuxmaniac**
#### Pre-Submission Checklist

- [x] Commit message has the format required by CONTRIBUTING guide
- [x] Commits are split per component (core, individual modules, libs, utils, ...)
- [x] Each component has a single commit (if not, squash them into one commit)
- [x] No commits to README files for modules (changes must be done to docbook files
in `doc/` subfolder, the README file is autogenerated)

#### Type Of Change
- [ ] Small bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)

#### Checklist:

- [ ] PR should be backported to stable branches
- [x] Tested changes locally

#### Description

This change introduces a new type of variable $xavi - eXtended Attribute Value Insensitive case.
It's like a $xavp but keys are case insensitive:

```
$xavi(WhatEver=>fOo) = 1;

if($xavi(whatever[0]=>foo) == 1) {
....
}
```

### CODE REVIEW

**Comment by miconda** [comment]
I could not spot anything on a quick look over the patch. If nobody else wants to still keep it here for review, I am fine to merge it.

**Comment by miconda** [comment]
I prefer to have separate functions, instead of going to some generic void* callbacks. Those are hard to track and prone to introduce bugs in existing functional code. The duplicated code should not be really big in size and having new dedicated functions for comparison, etc ... will result also in extra size of the overall code.

I actually looked at reusing the xavp code when I implementing xavu recently and the code got a lot of messy IF-ELSE conditions everywhere, functions with extra parameters to indicate what type of variable, etc ... so I reverted and just made dedicated functions, everything afterwards being easier to follow.

**Comment by henningw** [comment]
I don't think the usage of function pointers is a big problem, was just an idea. If you need to pass to many "switch" type of variables, it gets hard to follow, sure.
The two functions xavi_add_value, xavi_add_value_after just operate on a global static list. This could be implemented type safe by just passing the list to it, without the need of void pointers.

**Comment by miconda** [comment]
In the case of the two functions you mention, then `_xavi_list_crt` is hidden in the c file (being declared static, since it is controlled in very special cases related transaction states, ...), so if those functions are used outside of the file, then they still need  wrappers to call a common function (for xavp and xavi) with the adequate list.

And my remark was targeting exactly this case, because the functions have like 10 effective lines of code. A wrapper function is like 3 lines of code, with a jump on the stack. So overall it is not a significant gain.

It makes sense to have a function for a small number of lines of code, if the function is used in a lot of places, but if it going to be used in 2-3 places, then it has to be a significant number of codes of lines to really worth it.

Of course, it can be seen as a personal preference, but I prefer code easier to follow/search/maintain instead of modularising every snippet of code.
